backend/tests/unit/test_graph_builder.py

```
@@ -0,0 +1,595 @@
+        
+        graph_data = generate_vis_js_graph(modules)
+        
+        # Should have 2 edges
+        assert len(graph_data.edges) == 2
+        
+        # Find required vs optional edges
+        required_edge = next(e for e in graph_data.edges if e["from"] == "B")
+        optional_edge = next(e for e in graph_data.edges if e["from"] == "C")
+        
+        # Required edge should be solid, optional should be dashed
+        assert required_edge.get("dashes") is None or required_edge.get("dashes") == False
```

Ruff (0.12.2)

Avoid equality comparisons to `False`; use `not required_edge.get("dashes"):` for false checks

Replace with `not required_edge.get("dashes")`

(E712)